### PR TITLE
add tidb-limitations.md

### PR DESCRIPTION
**Pull request by tangenta**
### What is changed, added or deleted? (Required)

Add tidb-limitations.md, including:
- Identifier length limitation
- Different object number limitation

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: tidb-limitations.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+title: TiDB 限制
```
Comment (lilin90) [inline tidb-limitations.md:2]:
```suggestion
title: TiDB 使用限制
```

File: tidb-limitations.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+category: reference
+summary: 本文列出 TiDB 上的部分限制
```
Comment (lilin90) [inline tidb-limitations.md:4]:
```suggestion
summary: 了解 TiDB 标识符长度的限制和不同对象的数量限制。
```

File: tidb-limitations.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+---
+
+## TiDB 限制
```
Comment (lilin90) [inline tidb-limitations.md:7]:
```suggestion
## TiDB 使用限制
```

File: tidb-limitations.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+| Index | 64 |
+| Constraint | 64 |
+| View | 64 |
+| Alias | 256 |
+| Sequence | 64 |
+| User Variable | 64 |
+
+### 不同对象的数量限制
+
+数据库数量和表数量：在 TiDB 中，数据库和表的数量限制在理论上取决于 ID 的长度，目前 TiDB 的数据库 ID 和 表 ID 长度均为 64 位，因此数据库和表的数量最大为 9223372036854775807 （即 `(1 << 63) - 1`）。在实际应用中，该数量不太可能达到上限。
```
Comment (lilin90) [inline tidb-limitations.md:29]:
```suggestion
- 数据库数量和表数量：在 TiDB 中，数据库和表的数量限制在理论上取决于 ID 的长度，目前 TiDB 的数据库 ID 和表 ID 长度均为 64 位，因此数据库和表的数量最大为 9223372036854775807（即 `(1 << 63) - 1`）。在实际应用中，该数量不太可能达到上限。
```

File: tidb-limitations.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+单表行数量：TiDB 的行 ID 的空间占用长度为 64 位，如果某表的主键不是整数类型的列，则理论上行数量最多为 9223372036854775807 （即 `(1 << 63) - 1`）行。当集群中存在多个 TiDB 实例时，每个 TiDB 会缓存一段行 ID 用于分配，随着 TiDB 实例的状态影响（例如重启、缓存刷新、使用了 `AUTO_INCREMENT` 等），可插入的行数量会有所减少。但是，在实际应用中，行数据量不太可能达到该上限。另一方面，当表的主键为整数类型列时，单表行数量会受到该列类型空间长度的限制。例如，对于主键列类型为 `TINYINT` 的表，行数量最多为 256 行。
```
Comment (lilin90) [inline tidb-limitations.md:31]:
```suggestion
- 单表行数量：TiDB 的行 ID 的空间占用长度为 64 位，如果某表的主键不是整数类型的列，则理论上行数量最多为 9223372036854775807（即 `(1 << 63) - 1`）行。当集群中存在多个 TiDB 实例时，每个 TiDB 会缓存一段行 ID 用于分配，随着 TiDB 实例的状态影响（例如重启、缓存刷新、使用了 `AUTO_INCREMENT` 等），可插入的行数量会有所减少。但是，在实际应用中，行数据量不太可能达到该上限。另一方面，当表的主键为整数类型列时，单表行数量会受到该列类型空间长度的限制。例如，对于主键列类型为 `TINYINT` 的表，行数量最多为 256 行。
```

File: tidb-limitations.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+单表内的列数量：TiDB 中单表的列数量大小限制为 512。
```
Comment (lilin90) [inline tidb-limitations.md:33]:
```suggestion
- 单表内的列数量：TiDB 中单表的列数量大小限制为 512。
```

File: tidb-limitations.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+单表内的分区数量：TiDB 中单表的分区数量大小限制为 1024。
```
Comment (lilin90) [inline tidb-limitations.md:35]:
```suggestion
- 单表内的分区数量：TiDB 中单表的分区数量大小限制为 1024。
```

File: tidb-limitations.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
+---
+title: TiDB 使用限制
```
Comment (scsldb) [inline tidb-limitations.md:2]: 这个文件我记得我写过了，为啥又搞一个呢？
@tangenta

Comment (tangenta) [reply tidb-limitations.md:2]: @scsldb 我看到了，那 close 掉这个 PR 吧

**Comment by tangenta** [comment]
Duplicated with #3166.

**Pull request closed without merging**